modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/210_logsdb_default_mapping.yml

```diff
@@ -0,0 +1,254 @@
+                "@timestamp":
+                  type: "date"
+                  format: "strict_date_optional_time"
+
+  - do:
+      indices.put_index_template:
+        name: "logsdb-index-template"
+        body:
+          index_patterns: ["logsdb"]
+          data_stream: {}
+          composed_of: ["logsdb-mappings"]
+
+  - do:
+      catch: bad_request
+      indices.create_data_stream:
+        name: "logsdb"
+
+  - match: { error.type: "mapper_parsing_exception" }
+  - match: { error.reason: "Failed to parse mapping: can't merge a non object mapping [host] with an object mapping" }
+
+---
+create logsdb data stream with host as text and name as double:
+  - requires:
+      cluster_features: ["mapper.keyword_normalizer_synthetic_source"]
+      reason: "Support for normalizer on keyword fields"
+
+  - do:
+      cluster.put_component_template:
+        name: "logsdb-mappings"
+        body:
+          template:
+            settings:
+              mode: "logsdb"
+            mappings:
+              properties:
+                host:
+                  type: "text"
+                  fields:
+                    name:
+                      type: "double"
+                "@timestamp":
+                  type: "date"
+                  format: "strict_date_optional_time"
+
+  - do:
+      indices.put_index_template:
+        name: "logsdb-index-template"
+        body:
+          index_patterns: ["logsdb"]
+          data_stream: {}
+          composed_of: ["logsdb-mappings"]
+
+  - do:
+      catch: bad_request
+      indices.create_data_stream:
+        name: "logsdb"
+
+  - match: { error.type: "mapper_parsing_exception" }
+  - match: { error.reason: "Failed to parse mapping: can't merge a non object mapping [host] with an object mapping" }
+
+---
+create logsdb data stream with timestamp object mapping:
+  - requires:
+      cluster_features: ["mapper.keyword_normalizer_synthetic_source"]
+      reason: "Support for normalizer on keyword fields"
+
+  - do:
+      cluster.put_component_template:
+        name: "logsdb-mappings"
+        body:
+          template:
+            settings:
+              mode: "logsdb"
+            mappings:
+              properties:
+                host:
+                  properties:
+                    name:
+                      type: "keyword"
+                "@timestamp":
+                  properties:
+                    date:
+                      type: "date"
+                      format: "strict_date_optional_time"
+
+  - do:
+      catch: bad_request
+      indices.put_index_template:
+        name: "logsdb-index-template"
+        body:
+          index_patterns: ["logsdb"]
+          data_stream: {}
+          composed_of: ["logsdb-mappings"]
+
+  - match: { error.type: "illegal_argument_exception" }
+  - match: { error.reason: "composable template [logsdb-index-template] template after composition with component templates [logsdb-mappings] is invalid" }
```

Comment:
This is not that informative.. any chance we can catch this and complain about @timestamp?

Reply:
The is just the top level exception message. There is another one saying
```
The timestamp field [@timestamp] does not exist in the data stream
```

I rpefer not to change this as part of this PR...this tests just capture whatever the status is now.

Reply:
This PR is about adding tests for what current expected behavior is. We can consider improving the error message in a follow up PR. I suspect this error is data stream related and not logsdb related. There is data stream logic in place around `@timestamp`.